SCClassLibrary/Common/Control/ServerNodeTree.sc

```diff
@@ -0,0 +1,198 @@
+ServerNodeTree {
+
+	var server;
+	var updateFunc, updater, <>refreshRate = 0.2;
+	var currentMsg, currentNodeTree = nil;
+	var <running = false;
+	var oscResp;
+	
+	*new { |server|
+		^super.new.init(server)
+	}
+	
+	init { |parentServer|
+		server = parentServer;
+	}
+
+	getNodeTree {
+		if(running) {
+			if(currentNodeTree.isNil) {
+				this.prFormatNodeTree;
+			};
+			^currentNodeTree
+		} {
+			^nil
+		};
+	}
+
+	run {
+		if(running.not) {
+			// Init osc receiver
+			oscResp = OSCFunc(
+				{ |msg|
+					// Only catch relevant messages
+					// (root node and controls enabled)
+					if((msg[1] == 1) and: { msg[2] == 0 }) {
+						this.prClearNodeTree;
```

Comment:
One thing may be an optimisation: up to a certain size of the incoming reply message, one could check if the message has stayed the same and only set the nodeTree to nil if it has changed.

I am not sure from what message size onwards it won't be a gain anymore.

Here is a benchmark for array comparison, maybe you can compare with `prFormatNodeTree`.
```supercollider
a = { [1, 2, 3, \a, \b].choose }.dup(100); b = a.copy; b.put(99, 10);
bench { 1000.do { a == b } };
```